controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -242,12 +244,14 @@ public void triggerChange(ApplicationId applicationId, Change change) {
      *
      * @param applicationId the application to trigger
      */
-    public void cancelChange(ApplicationId applicationId) {
+    public void cancelChange(ApplicationId applicationId, boolean keepApplicationChange) {
         applications().lockOrThrow(applicationId, application -> {
-            deploymentQueue.removeJobs(application.id());
             applications().store(application.withChange(application.change().application()
-                                                                .map(Change::of)
-                                                                .orElse(Change.empty())));
+                                                                   .map(Change::of)
+                                                                   .filter(change -> keepApplicationChange)
+                                                                   .orElse(Change.empty())));
+            if ( ! applications().require(applicationId).change().isPresent())
+                deploymentQueue.removeJobs(application.id());
```

Comment:
Seems wrong to cancel the job if if there's still an application change rolling.